### PR TITLE
Add Unix Timestamp value type

### PR DESCRIPTION
**Pull request by MattDelac**
**What this PR does / why we need it**:
As discussed in Slack, so far, we don't want to support datetime dtypes in general. Therefore, so far we would need to convert any datetime data types into a unix timestamp and store it as an INT in feast.

While converting in unix timestamp is not an issue, it makes it hard to differentiate the features that are int vs timestamp when looking at a FeatureView

Thus this PR aims to introduce a new datatype "Unix Timestamp" that beyond the scene would be an INT. For the end user, this will make it easier understand a FeatureView.
It also add a potential possibility to do things differently with Unix Timestamp features at retrieval time.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Add UnixTimestamp valueType that is an Int64 under the hood
```

### CODE REVIEW

File: sdk/python/tensorflow_metadata/proto/v0/path_pb2.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 # Generated by the protocol buffer compiler.  DO NOT EDIT!
 # source: tensorflow_metadata/proto/v0/path.proto
-"""Generated protocol buffer code."""
```
Comment (woop) [inline sdk/python/tensorflow_metadata/proto/v0/path_pb2.py:4]: This is weird. Was this manually edited?

**Comment by woop** [comment]
Hey @MattDelac, thanks for the contribution!

Overall it looks good to me. One thing that is a glaring omissions (not with your PR but with our code) is that we're not testing all the different types. Regardless, we can add that test a bit later. I've added a note to add a [test here](https://github.com/feast-dev/feast/issues/1496).

I think the only thing missing here is an update to our documentation. You may need to update the specs to document the new type: https://github.com/feast-dev/feast/tree/master/docs/specs

File: sdk/python/tensorflow_metadata/proto/v0/path_pb2.py (hunk 1 above)
Comment (MattDelac) [reply sdk/python/tensorflow_metadata/proto/v0/path_pb2.py:4]: Don't think I edited anything manually for a pb2 file
I can start from a fresh master and see if that happens again after compiling protos ...

**Comment by MattDelac** [comment]
> One thing that is a glaring omissions (not with your PR but with our code) is that we're not testing all the different types.

Yes that was my surprise that none of the tests had to be changed

> I've added a note to add a test here.

Sounds good. I let you fix it later 👌 

> I think the only thing missing here is an update to our documentation

Good catch ! Will look into the doc

**Comment by MattDelac** [comment]
> I think the only thing missing here is an update to our documentation. You may need to update the specs to document the new type: https://github.com/feast-dev/feast/tree/master/docs/specs

Done. The docs are now up to date. Let me know if I missed another one.

File: sdk/python/feast/type_map.py [hunk 2]
```diff
@@ -252,6 +252,18 @@ def _python_value_to_proto_value(feast_value_type, value) -> ProtoValue:
                 )
             )
 
+        if feast_value_type == ValueType.UNIX_TIMESTAMP_LIST:
```
Comment (woop) [inline sdk/python/feast/type_map.py:255]: Should the SCALAR type also be handled below?

Comment (MattDelac) [reply sdk/python/feast/type_map.py:255]: True.

Done

**Review by woop — APPROVED** [review]

**Pull request merged**